tutor/src/helpers/analytics.js

```diff
@@ -58,7 +58,8 @@ const Translators = {
 
   // Task steps are viewed by both teacher and student with no difference in params
   viewTaskStep({ courseId }) {
-    const role = Courses.get(courseId).primaryRole.type;
+    const course = Courses.get(courseId);
+    const role = course ? course.primaryRole.type : 'unknown';
```

Comment:
Hmm not `currentRole` here?

Reply:
no, primaryRole will give priority to teacher role if present, otherwise student.  for this case we just want to log if a teacher or student is viewing the task step so the analytics make sense